src/datamodel/quests/quest_new.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::rc::Rc;
+
+#[derive(Debug, PartialEq)]
+pub enum Vote { Success, Failed, }
+
+pub struct QuestObjective {
+    pub num_players: u32,
+    pub required_successes: u32,
+}
+
+pub struct QuestNew {}
+
+impl QuestNew {
+    pub fn new(members: Vec<&'static str>, objective: Rc<QuestObjective>) -> OnGoingQuest {
+        let members: HashMap<&'static str, bool> = members.into_iter().map(|m| (m, false)).collect();
+        OnGoingQuest { 
+                objective,
+                members,
+                voting_counter: 0,
+                success_votes: 0,
+        }
+    }
+}
+
+pub struct OnGoingQuest {
+    members: HashMap<&'static str, bool>,
+    voting_counter: u32,
+    objective: Rc<QuestObjective>,
+    success_votes: u32,
+}
+
+impl OnGoingQuest {
+
+    pub fn add_vote(&mut self, player: &str, vote: Vote) {
+        if self.members[player] == true {
```

Comment:
Can't we use the boolean value instead of adding the extra comparison?

```suggestion
        if self.members[player] {
```